### PR TITLE
nats: add a reply param to nats_publish and expose as nats_publish_request in KEMI

### PR DESCRIPTION
**Pull request by seven1240**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description

I added `nats_publish_request` which calls the `natsConnection_PublishRequestString` to send a request to the NATS server with a `reply` string so the response can be easily replied back.

I tested in Lua and the native route, not sure if it's the right way to implement. I can update docs if it sounds good.

Thanks.

### CODE REVIEW

**Comment by seven1240** [comment]
@eschmidbauer can you take a look if this one makes sense? Thanks.

File: src/modules/nats/nats_mod.c [hunk 1]
```diff
@@ -60,6 +60,8 @@ static param_export_t params[] = {
 static cmd_export_t cmds[] = {
 	{"nats_publish", (cmd_function)w_nats_publish_f,
 		  2, fixup_publish_get_value, fixup_publish_get_value_free, ANY_ROUTE},
```
Comment (eschmidbauer) [inline src/modules/nats/nats_mod.c:62]: are these two lines still needed?

Comment (seven1240) [reply src/modules/nats/nats_mod.c]: I removed and rebased to the latest master, seems fine for me. Thanks.

**Comment by eschmidbauer** [comment]
does this change require a 3rd parameter to `nats_publish()` ?

**Comment by seven1240** [comment]
> does this change require a 3rd parameter to `nats_publish()` ?

No, the 3rd one is optional.

```
route[nats] {
    nats_publish("cn.xswitch.ctrl.route", 'hello');
    nats_publish("cn.xswitch.ctrl.route", 'hello', 'blah');
}
```

**Comment by eschmidbauer** [comment]
looks good to me. can you update the documentation to reflect the new parameter?

**Comment by henningw** [comment]
Thanks for the PR. Lets wait a few more days for more feedback, and then merge it.

**Pull request merged**